editoast/src/views/timetable.rs

```diff
@@ -424,34 +513,124 @@ async fn conflicts(
     )
     .await?;
 
-    // 2. Build core request
+    let mut paced_trains_ts = vec![];
+    for paced_train in paced_trains.clone() {
+        let occurrences =
+            (paced_train.duration.num_seconds() / paced_train.step.num_seconds()) as usize;
+        let step = paced_train.step;
+        let mut start_time = paced_train.start_time;
+
+        let mut all_occurrences = Vec::with_capacity(occurrences);
+
+        for _ in 0..occurrences {
+            let mut first_occurrence = paced_train.clone().into_first_occurrence();
+            first_occurrence.start_time = start_time;
+            all_occurrences.push(first_occurrence);
+            start_time += step;
+        }
+
+        paced_trains_ts.extend(all_occurrences);
+    }
+
+    let paced_train_simulations = train_simulation_batch(
+        &mut db_pool.get().await?,
+        valkey_client.clone(),
+        core_client.clone(),
+        &paced_trains_ts,
+        &infra,
+        electrical_profile_set_id,
+    )
+    .await?;
+
+    let (conflict_detection_request, map_string_to_id) = build_conflict_core_request(
+        infra,
+        trains,
+        train_simulations,
+        paced_trains_ts,
+        paced_train_simulations,
+    );
+
+    // 3. Call core
+    let conflict_detection_response = conflict_detection_request.fetch(&core_client).await?;
+    let conflicts = conflict_detection_response.conflicts;
+    let conflicts_response: Result<Vec<Conflict>> = conflicts
+        .into_iter()
+        .map(|conflict| Conflict::into_conflict_response(conflict, map_string_to_id.clone()))
+        .collect();
+    Ok(Json(conflicts_response?))
+}
+
+fn build_conflict_core_request(
+    infra: Infra,
+    trains: Vec<TrainSchedule>,
+    train_simulations: Vec<(SimulationResponse, PathfindingResult)>,
+    paced_trains_ts: Vec<TrainSchedule>,
+    paced_train_simulations: Vec<(SimulationResponse, PathfindingResult)>,
+) -> (ConflictDetectionRequest, HashMap<String, TrainId>) {
+    let mut map_string_to_id: HashMap<String, TrainId> = HashMap::new();
```

Comment:
As said earlier, I think we can get rid of the map since all `TrainId`s are embedded in the request and will provided back in the response.